test/gateway/eip20_gateway/progress_revert_stake.js

```diff
@@ -0,0 +1,443 @@
+      ),
+      'Message on source must be DeclaredRevocation.',
+    );
+
+  });
+
+  it('should fail when message status of inbox at target is progressed', async function () {
+
+    let paramsData = progressedProofData.gateway.stake.params;
+
+    revertStakeParams = {
+      amount: new BN(paramsData.amount, 16),
+      beneficiary: paramsData.beneficiary,
+      gasPrice: new BN(paramsData.gasPrice, 16),
+      gasLimit: new BN(paramsData.gasLimit, 16),
+      nonce: new BN(paramsData.nonce, 16),
+      hashLock: paramsData.hashLock,
+      messageHash: progressedProofData.gateway.stake.return_value.returned_value.messageHash_,
+      staker: paramsData.staker,
+      rlpParentNodes: progressedProofData.co_gateway.progress_mint.proof_data.storageProof[0].serializedProof,
+      blockHeight: new BN(progressedProofData.co_gateway.progress_mint.proof_data.block_number,16),
+      gatewayAddress: progressedProofData.contracts.gateway,
+      storageRoot: progressedProofData.co_gateway.progress_mint.proof_data.storageHash,
+    };
+
+    await setup(revertStakeParams);
+
+    await mockToken.transfer(gateway.address, revertStakeParams.amount, { from: accounts[0] });
+    await baseToken.transfer(gateway.address, bountyAmount.muln(2.5), { from: accounts[0] });
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    await Utils.expectRevert(
+      gateway.progressRevertStake(
+        revertStakeParams.messageHash,
+        revertStakeParams.blockHeight,
+        revertStakeParams.rlpParentNodes,
+      ),
+      'Merkle proof verification failed.',
+    );
+
+  });
+  
+  it('should pass when message status of inbox at target is revoked', async function () {
+
+    await gateway.setStorageRoot(revertStakeParams.blockHeight, revertStakeParams.storageRoot);
+
+    let result = await gateway.progressRevertStake.call(
+      revertStakeParams.messageHash,
+      revertStakeParams.blockHeight,
+      revertStakeParams.rlpParentNodes,
+    );
+
+    assert.strictEqual(
+      result.staker_,
+      revertStakeParams.staker,
+      `Staker address must be equal to ${revertStakeParams.staker}.`,
+    );
+
+    assert.strictEqual(
+      result.amount_.eq(revertStakeParams.amount),
+      true,
+      `Stake amount ${result.amount_.toString(10)} must be equal to ${revertStakeParams.amount.toString(10)}.`,
+    );
+
+    assert.strictEqual(
+      result.stakerNonce_.eq(revertStakeParams.nonce),
+      true,
+      `Staker nonce ${result.stakerNonce_.toString(10)} must be equal to ${revertStakeParams.nonce.toString(10)}.`,
+    );
+
+    let tx = await gateway.progressRevertStake(
+      revertStakeParams.messageHash,
+      revertStakeParams.blockHeight,
+      revertStakeParams.rlpParentNodes,
+    );
+
+    assert.equal(
+      tx.receipt.status,
+      1,
+      "Receipt status is unsuccessful",
+    );
+
+  });
+
+  it('should emit `StakeReverted` event ', async function () {
```

Comment:
Some have extra spaces again 🧐
```suggestion
  it('should emit `StakeReverted` event', async function () {
```